DevAcademy/ObservableObject/PlacesObservableObject.swift

```diff
@@ -21,14 +35,39 @@ final class PlacesObservableObject: ObservableObject {
             switch result {
             case .success(let places):
                 DispatchQueue.main.async {
-                    self.places = places.features
+                    self.rawPlaces = places.features
                 }
             case .failure(let error):
                 print(error)
             }
         }
     }
     
+    func set(place: Place, favorite: Bool) {
+        if favorite {
+            if !isFavorite(place: place) { favorites?.append(place.attributes.ogcFid) }
+        } else {
+            favorites?.removeAll { $0 == place.attributes.ogcFid }
+        }
+    }
+    
+    func updatePlaces() {
+        places.append(contentsOf: rawPlaces)
+        guard let favorites = favorites else { return }
+        
+        for favoriteID in favorites {
+            if let index = places.firstIndex(where: { $0.attributes.ogcFid == favoriteID }) {
+                let item = places.remove(at: index)
+                places.insert(item, at: 0)
+            }
+        }
+    }
```

Comment:
```suggestion
    func updatePlaces() {
        var newPlaces = rawPlaces
        guard let favorites = favorites else { 
            places = newPlaces
            return
        }
        
        for favoriteID in favorites {
            if let index = newPlaces.firstIndex(where: { $0.attributes.ogcFid == favoriteID }) {
                let item = newPlaces.remove(at: index)
                newPlaces.insert(item, at: 0)
            }
        }
        places = newPlaces
    }
```

První věc, doporučuji pokud aktualizujete @Published proměnnou - vždycky si udělejte kopii a potom tuto proměnnou přepište. Vyhnete se opakovanému generování zbytečných eventů.

Druhý problém je, že vy vlastně do places neustále přidáváte nová místa - tím pádem se vám stejné místo v tabulce objeví tolikrát, kolikrát zavoláte funkci updatePlaces :)

Reply:
Super tip, upravil jsem to dle zadání (jen jsem ho maličko pozměnil, aby to fungovalo - jak řazení, tak i zobrazení v bottomSheetu) :-)

Reply:
Ještě jsem přidal domácí úkol ze včerejška :)